src/Mjolnir.ts

```diff
@@ -81,6 +81,11 @@ export class Mjolnir {
 
     public readonly policyListManager: PolicyListManager;
 
+    /**
+     * Members of the moderator room and others who should not be banned, ACL'd etc.
+     */
+    public moderators: string[];
```

Comment:
Users who are invited/joined after the bot starts up would get banned. Adding a cache would help ensure we don't miss anyone. We'd invalidate the cache on two conditions I think:
1. Every 60 minutes.
2. Whenever we see an `m.room.member` event in the management room. (we don't have to inspect the event: seeing it fly by should be enough to invalidate&repopulate)

For deployments like matrix.org this may mean we pick up on displayname changes quite often, but I'd consider that a feature for keeping the cache active.